src/main/kotlin/interactor/GetTop10CountriesWithHighTaxOnCarbonatedDrinks.kt

```diff
@@ -0,0 +1,23 @@
+package interactor
+import model.DrinksPrices
+import model.CityEntity
+import java.util.*
+
+class GetTop10CountriesWithHighTaxOnCarbonatedDrinks ( private val dataSource: CostOfLivingDataSource){
+    fun execute(cities:List<CityEntity>): List<Pair<String, Float?>> {
+        return cities
+            .asSequence()
+            .filter(::excludeNullDrinkPriceAndNegativePriceAndLowQualityData)
+            .distinctBy { it.country }
+            .sortedByDescending { it.drinksPrices.cokePepsiAThirdOfLiterBottleInRestaurants }
+            .take(10)
```

Comment:
```suggestion
    fun execute(
        cities:List<CityEntity>,
        limit: Int
): List<Pair<String, Float?>> {
        return cities
            .asSequence()
            .filter(::excludeNullDrinkPriceAndNegativePriceAndLowQualityData)
            .distinctBy { it.country }
            .sortedByDescending { it.drinksPrices.cokePepsiAThirdOfLiterBottleInRestaurants }
            .take(limit)
```

Declare a limit variable, and use it instead of hardcoded value, to make it a pure function.
You should change the test code for that.

Reply:
My Question it's only want top 10 why I need to change it ?

Reply:
To make it pure function, and as Braeq says:"Avoid any hard coded values"

Reply:
![Screenshot_20230212-121241_GitHub.png](https://user-images.githubusercontent.com/92039633/218305051-00f49658-f75e-4603-b36a-8abef5252cdd.png)

Reply:
okay ,thanks for make it clear

Reply:
Thanks for Ur effort to do it